app/models/article.rb

```diff
@@ -27,6 +27,9 @@ class Article < ApplicationRecord
 
   # The date that we began limiting the number of user mentions in an article.
   MAX_USER_MENTION_LIVE_AT = Time.utc(2021, 4, 7).freeze
+  # The commented variable below is the image markdown regex for images stored in the local public folder.
+  # IMG_MARKDOWN_REGEX = %r{!\[.*\]\(#{ApplicationConfig["APP_PROTOCOL"]}#{ApplicationConfig["APP_DOMAIN"]}\/uploads\/articles\/(.*)\)}.freeze
+  IMG_MARKDOWN_REGEX = %r{!\[.*\]\(#{ApplicationConfig["APP_PROTOCOL"]}#{ApplicationConfig["AWS_BUCKET_NAME"]}.s3.amazonaws.com\/uploads\/articles(.*)\)}.freeze
```

Comment:
This line is super long, it's odd that rubocop didn't flag it during "git commit": 

Can you run `bundle exec rubocop -a app/models/article.rb` and fix the remaining issue(s)?

The `Layout/LineLength` can be quieted with: `IMG_MARKDOWN_REGEX = %r{...} # rubocop:disable Layout/LineLength` (the comment has to be inline).

That said, I don't think we should hardcode the path here, you can get it from `ArticleImageUploader.new.store_dir`

Reply:
Yes I wanted to ask about how to tackle the hardcode url. Thanks for the info. If I am understanding it correctly, the current changes should work only for DEV, specifically for the images in articles, since the regex takes AWS values from the config. I think I could code the regex to take a storage_host and storage_dir from the config. Imgproxy images won't be detected however, their urls are structured differently, and depending on how they're stored, it might need separate deletion logic.

Reply:
Cool! I like that we're moving in steps instead of trying to do it all at once! Thank you!

Reply:
I fixed the regex to take the path from store_dir. The regex will still detect only images stored in aws s3 though. Would it be a good idea to add a config variable like `AWS_S3_STORAGE_HOST = s3.amazonaws.com`, or perhaps create a few non aws specific config variables?

Reply:
I'm not sure how we'll go about deleting images stored on imgproxy and as we said, this PR is about only DEV's Cloudinary images. I don't think we should add anything else